[pulseaudio] Offline Pulseaudio Bridge triggers throwing of exceptions

```
2020-10-23 18:35:41.455 [ERROR] [pulseaudio.internal.PulseaudioClient] - No route to host (Host unreachable)
java.net.NoRouteToHostException: No route to host (Host unreachable)
	at java.net.PlainSocketImpl.socketConnect(Native Method) ~[?:1.8.0_265]
	at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:350) ~[?:1.8.0_265]
	at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:206) ~[?:1.8.0_265]
	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:188) ~[?:1.8.0_265]
	at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392) ~[?:1.8.0_265]
	at java.net.Socket.connect(Socket.java:607) ~[?:1.8.0_265]
	at java.net.Socket.connect(Socket.java:556) ~[?:1.8.0_265]
	at java.net.Socket.<init>(Socket.java:452) ~[?:1.8.0_265]
	at java.net.Socket.<init>(Socket.java:229) ~[?:1.8.0_265]
	at org.openhab.binding.pulseaudio.internal.PulseaudioClient.connect(PulseaudioClient.java:611) ~[?:?]
	at org.openhab.binding.pulseaudio.internal.PulseaudioClient.checkConnection(PulseaudioClient.java:599) ~[?:?]
	at org.openhab.binding.pulseaudio.internal.PulseaudioClient.sendRawRequest(PulseaudioClient.java:555) ~[?:?]
	at org.openhab.binding.pulseaudio.internal.PulseaudioClient.listModules(PulseaudioClient.java:166) ~[?:?]
	at org.openhab.binding.pulseaudio.internal.PulseaudioClient.update(PulseaudioClient.java:143) ~[?:?]
	at org.openhab.binding.pulseaudio.internal.PulseaudioClient.<init>(PulseaudioClient.java:131) ~[?:?]
	at org.openhab.binding.pulseaudio.internal.handler.PulseaudioBridgeHandler.lambda$1(PulseaudioBridgeHandler.java:135) ~[?:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_265]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_265]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) [?:1.8.0_265]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) [?:1.8.0_265]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_265]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_265]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_265]
2020-10-23 18:35:41.476 [WARN ] [mmon.WrappedScheduledExecutorService] - Scheduled runnable ended with an exception: 
java.lang.NullPointerException: null
	at org.openhab.binding.pulseaudio.internal.PulseaudioClient.sendRawRequest(PulseaudioClient.java:558) ~[?:?]
	at org.openhab.binding.pulseaudio.internal.PulseaudioClient.listModules(PulseaudioClient.java:166) ~[?:?]
	at org.openhab.binding.pulseaudio.internal.PulseaudioClient.update(PulseaudioClient.java:143) ~[?:?]
	at org.openhab.binding.pulseaudio.internal.PulseaudioClient.<init>(PulseaudioClient.java:131) ~[?:?]
	at org.openhab.binding.pulseaudio.internal.handler.PulseaudioBridgeHandler.lambda$1(PulseaudioBridgeHandler.java:135) ~[?:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) ~[?:1.8.0_265]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[?:1.8.0_265]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) ~[?:1.8.0_265]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) ~[?:1.8.0_265]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_265]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_265]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_265]
```

## Expected Behavior
Log a single message about the device being offline (at most) and set the device state to offline, since an offline state seems something that could be expected of a pulseaudio bridge

## Current Behavior
The connection request to an offline pulseaudio client causes a NoRouteToHostException to be thrown. 

## Possible Solution
Catch the exception and add a logger statement

## Steps to Reproduce (for Bugs)
- set up a pulseaudio server on some linux box.
- enable its cli-protocol and set up the bridge and perhaps a sink item in openhab
- confirm the creation and states of the items and bridge in the log
- take the pulseaudio server offline

## Context 
In my setup i have 3 thin clients in different rooms running linux and pulseaudio. Sometimes these devices are shut off or in sleep mode to conserve energy.
Whenever this is the case and the pulseaudio binding does *something* periodically, i get a long stack trace message in my logs resulting of the route to the host not being found. 
I think this exception should be handled more gracefully, so as not to spam the logs.

## Environment
* Version used: (openHAB 2.5.9-1 on openhabian)
* Environment name and version: pulseaudio servers in version 12.x and 13.x

--- a/bundles/org.openhab.binding.pulseaudio/src/main/java/org/openhab/binding/pulseaudio/internal/PulseaudioClient.java
+++ b/bundles/org.openhab.binding.pulseaudio/src/main/java/org/openhab/binding/pulseaudio/internal/PulseaudioClient.java
@@ -17,6 +17,7 @@ import static org.openhab.binding.pulseaudio.internal.PulseaudioBindingConstants
 import java.io.IOException;
 import java.io.InputStream;
 import java.io.PrintStream;
+import java.net.NoRouteToHostException;
 import java.net.Socket;
 import java.net.SocketException;
 import java.net.SocketTimeoutException;
@@ -132,7 +133,7 @@ public class PulseaudioClient {
     }
 
     public boolean isConnected() {
-        return client.isConnected();
+        return client != null ? client.isConnected() : false;
     }
 
     /**
@@ -539,14 +540,16 @@ public class PulseaudioClient {
 
     private void sendRawCommand(String command) {
         checkConnection();
-        try {
-            PrintStream out = new PrintStream(client.getOutputStream(), true);
-            logger.trace("sending command {} to pa-server {}", command, host);
-            out.print(command + "\r\n");
-            out.close();
-            client.close();
-        } catch (IOException e) {
-            logger.error("{}", e.getLocalizedMessage(), e);
+        if (client != null) {
+            try {
+                PrintStream out = new PrintStream(client.getOutputStream(), true);
+                logger.trace("sending command {} to pa-server {}", command, host);
+                out.print(command + "\r\n");
+                out.close();
+                client.close();
+            } catch (IOException e) {
+                logger.error("{}", e.getLocalizedMessage(), e);
+            }
         }
     }
 
@@ -554,41 +557,43 @@ public class PulseaudioClient {
         logger.trace("_sendRawRequest({})", command);
         checkConnection();
         String result = "";
-        try {
-            PrintStream out = new PrintStream(client.getOutputStream(), true);
-            out.print(command + "\r\n");
-
-            InputStream instr = client.getInputStream();
-
+        if (client != null) {
             try {
-                byte[] buff = new byte[1024];
-                int retRead = 0;
-                int lc = 0;
-                do {
-                    retRead = instr.read(buff);
-                    lc++;
-                    if (retRead > 0) {
-                        String line = new String(buff, 0, retRead);
-                        // System.out.println("'"+line+"'");
-                        if (line.endsWith(">>> ") && lc > 1) {
-                            result += line.substring(0, line.length() - 4);
-                            break;
+                PrintStream out = new PrintStream(client.getOutputStream(), true);
+                out.print(command + "\r\n");
+
+                InputStream instr = client.getInputStream();
+
+                try {
+                    byte[] buff = new byte[1024];
+                    int retRead = 0;
+                    int lc = 0;
+                    do {
+                        retRead = instr.read(buff);
+                        lc++;
+                        if (retRead > 0) {
+                            String line = new String(buff, 0, retRead);
+                            // System.out.println("'"+line+"'");
+                            if (line.endsWith(">>> ") && lc > 1) {
+                                result += line.substring(0, line.length() - 4);
+                                break;
+                            }
+                            result += line.trim();
                         }
-                        result += line.trim();
-                    }
-                } while (retRead > 0);
-            } catch (SocketTimeoutException e) {
-                // Timeout -> as newer PA versions (>=5.0) do not send the >>> we have no chance
-                // to detect the end of the answer, except by this timeout
+                    } while (retRead > 0);
+                } catch (SocketTimeoutException e) {
+                    // Timeout -> as newer PA versions (>=5.0) do not send the >>> we have no chance
+                    // to detect the end of the answer, except by this timeout
+                } catch (IOException e) {
+                    logger.error("Exception while reading socket: {}", e.getMessage());
+                }
+                instr.close();
+                out.close();
+                client.close();
+                return result;
             } catch (IOException e) {
-                logger.error("Exception while reading socket: {}", e.getMessage());
+                logger.error("{}", e.getLocalizedMessage(), e);
             }
-            instr.close();
-            out.close();
-            client.close();
-            return result;
-        } catch (IOException e) {
-            logger.error("{}", e.getLocalizedMessage(), e);
         }
         return result;
     }
@@ -612,6 +617,8 @@ public class PulseaudioClient {
             client.setSoTimeout(500);
         } catch (UnknownHostException e) {
             logger.error("unknown socket host {}", host);
+        } catch (NoRouteToHostException e) {
+            logger.error("no route to host {}", host);
         } catch (SocketException e) {
             logger.error("{}", e.getLocalizedMessage(), e);
         }

